internal/textfile/read_test.go

```diff
@@ -3,25 +3,40 @@ package textfile
 import (
 	"bytes"
 	"encoding/hex"
+	"io/ioutil"
+	"os"
 	"testing"
-
-	"github.com/restic/restic/internal/fs"
 )
 
-func writeTempfile(t testing.TB, data []byte) (fs.File, func()) {
-	f, removeTempfile := fs.TestTempFile(t, "restic-test-textfile-read-")
+// writeTempfile writes data to a new temporary file and returns its name
+// and a callback that removes it.
+func writeTempfile(t testing.TB, data []byte) (string, func()) {
+	t.Helper()
 
-	_, err := f.Write(data)
+	f, err := ioutil.TempFile("", "restic-test-textfile-read-")
 	if err != nil {
 		t.Fatal(err)
 	}
+	name := f.Name()
+
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
 
-	err = f.Close()
+	_, err = f.Write(data)
```

Comment:
Shouldn't the test also delete the tempfile if the write fails? Maybe add a add a `shouldDelete` flag and check that in the deferred function?

Reply:
Good catch. I found a different way of doing this.